internal/pginfer/nullability.go

```diff
@@ -24,10 +24,10 @@ func isColNullable(query *ast.SourceQuery, plan Plan, out string, column pg.Colu
 		// try below
 	}
 
-	// A plain select query with no joins where the column comes from a table and
-	// has a not-null constraint. Not full proof because of cross-join with comma
-	// syntax.
-	if plan.Type == PlanResult &&
+	// A plain select query (possibly with a LIMIT clause) with no joins where
```

Comment:
I think this is valid. If postgres can trace the column back to a table and there's no join, we're probably good.